pkgs/build-support/trivial-builders/default.nix

```diff
@@ -367,6 +367,7 @@ rec {
     , text
     , runtimeShell ? runtimeShell'
     , runtimeInputs ? [ ]
+    , runtimeEnv ? null
```

Comment:
This reminds me of `makeWrapper`, which provides the same functionality via `--set <VAR> <VALUE>`. How about we make this function internally use `makeWrapper` instead? We can then also use `--prefix PATH : ${makeBinPath runtimeInputs}`.

Reply:
`makeWrapper` doesn't support setting values to lists or maps. I also think it's nicer to keep the written script contained to one file.

Reply:
Fair enough, we might want to reconsider this in the future though, especially if the need for more makeWrapper-like features comes up.